crypto/s2n_hash.h

```diff
@@ -37,43 +37,28 @@ typedef enum {
     S2N_HASH_ALGS_COUNT
 } s2n_hash_algorithm;
 
-/* The low_level_digest stores all OpenSSL structs that are alg-specific to be used with OpenSSL's low-level hash API's. */
-union s2n_hash_low_level_digest {
-    MD5_CTX md5;
-    SHA_CTX sha1;
-    SHA256_CTX sha224;
-    SHA256_CTX sha256;
-    SHA512_CTX sha384;
-    SHA512_CTX sha512;
-    struct {
-        MD5_CTX md5;
-        SHA_CTX sha1;
-    } md5_sha1;
-};
-
 /* The evp_digest stores all OpenSSL structs to be used with OpenSSL's EVP hash API's. */
 struct s2n_hash_evp_digest {
     struct s2n_evp_digest evp;
     /* Always store a secondary evp_digest to allow resetting a hash_state to MD5_SHA1 from another alg. */
     struct s2n_evp_digest evp_md5_secondary;
 };
 
-/* s2n_hash_state stores the s2n_hash implementation being used (low-level or EVP),
- * the hash algorithm being used at the time, and either low_level or high_level (EVP) OpenSSL digest structs.
+/* s2n_hash_state stores the state and a reference to the implementation being used.
+ * Currently only EVP hashing is supported, so the only state are EVP_MD contexts.
  */
 struct s2n_hash_state {
     const struct s2n_hash *hash_impl;
     s2n_hash_algorithm alg;
     uint8_t is_ready_for_input;
     uint64_t currently_in_hash;
     union {
-        union s2n_hash_low_level_digest low_level;
         struct s2n_hash_evp_digest high_level;
     } digest;
 };
 
-/* The s2n hash implementation is abstracted to allow for separate implementations, using
- * either OpenSSL's low-level algorithm-specific API's or OpenSSL's EVP API's.
+/* The s2n hash implementation is abstracted to allow for separate implementations.
+ * Currently the only implementation uses the EVP APIs.
  */
```

Comment:
I know we only have one implementation now, but it seemed like a bit of a waste to rip out the mechanism to handle multiple implementations. We might need another implementation someday, and I don't think this abstraction adds much complexity.